### ISSUE TITLE
svr4pkg fails to detect it installed packages

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29280]
From @dothebart on 2016-03-04T15:54:23Z
##### Issue Type:
- Bug Report
##### Component Name:

svr4pkg
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

none.
##### Environment:

Host: Debian Jessie x86_64 with upstream ansible packages installed
Solaris 11R3 installed via http://www.resilvered.com/2014/02/solaris-vagrant-packer-and-base-box.html
##### Summary:

first time installing the CSWcommon package as in the examples at http://docs.ansible.com/ansible/svr4pkg_module.html:

```
# Install a package from an already copied file
- svr4pkg: name=CSWcommon src=/tmp/cswpkgs.pkg state=present
# Install a package directly from an http site
- svr4pkg: name=CSWpkgutil src=http://get.opencsw.org/now state=present zone=current
```
##### Steps To Reproduce:

Run playbook first time - works. Run playbook second time, fails to detect that step is already done.

running `pkgrm CSWpkgutil` to get the system back into the original state makes it work again.
##### Expected Results:

Should skip successfull states
##### Actual Results:

```
TASK [solaris11-build-host : svr4pkg] ******************************************
task path: /local/home/willi/src/buildsystem/ansible/roles/solaris11-build-host/tasks/main.yml:5
<192.168.173.92> ESTABLISH SSH CONNECTION FOR USER: jenkins
<192.168.173.92> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o Port=4444 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=jenkins -o ConnectTimeout=10 -o ControlPath=/local/home/willi/.ansible/cp/ansible-ssh-%h-%p-%r -tt 192.168.173.92 '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1457106819.39-197850102934629 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1457106819.39-197850102934629 )" )'
<192.168.173.92> PUT /tmp/tmpt_AUYc TO /export/home/jenkins/.ansible/tmp/ansible-tmp-1457106819.39-197850102934629/svr4pkg
<192.168.173.92> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o Port=4444 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=jenkins -o ConnectTimeout=10 -o ControlPath=/local/home/willi/.ansible/cp/ansible-ssh-%h-%p-%r '[192.168.173.92]'
<192.168.173.92> ESTABLISH SSH CONNECTION FOR USER: jenkins
<192.168.173.92> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o Port=4444 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=jenkins -o ConnectTimeout=10 -o ControlPath=/local/home/willi/.ansible/cp/ansible-ssh-%h-%p-%r -tt 192.168.173.92 '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-oihnxxekgjykotumstzcfymenhlmbqvi; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /export/home/jenkins/.ansible/tmp/ansible-tmp-1457106819.39-197850102934629/svr4pkg; rm -rf "/export/home/jenkins/.ansible/tmp/ansible-tmp-1457106819.39-197850102934629/" > /dev/null 2>&1'"'"'"'"'"'"'"'"''"'"''
fatal: [buildSolaris11]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"category": false, "name": "CSWpkgutil", "proxy": null, "response_file": null, "src": "http://get.opencsw.org/now", "state": "present", "zone": "current"}, "module_name": "svr4pkg"}, "name": "CSWpkgutil", "state": "present"}


```

Copied from original issue: ansible/ansible-modules-extras#1792

**Comment by ansibot** [comment]
From @akamez on 2016-03-04T15:54:23Z

Hi, if you want a temporary fix while the module is patched

find your module file, mine was here:

`/usr/lib/python2.6/site-packages/ansible/modules/extras/packaging/os/svr4pkg.py`

Then edit the file as follows :

```
231c231
<     if rc not in (0, 2, 10, 20):
---
>     if rc not in (0, 2, 10, 20, None): # added a return code  of  None if no functions need running
```

**Comment by ansibot** [comment]
From @ceving on 2016-03-04T15:54:23Z

Same problem here with Solaris 10 and Ansible 2.1.1.0.

**Comment by ansibot** [comment]
From @brontitall on 2016-03-04T15:54:23Z

Hey, I'm afraid I do not currently use ansible, and have not for some years. Although I wrote this code some time ago, I no longer have the environment or knowledge to update it.

**Comment by ansibot** [comment]
From @ceving on 2016-03-04T15:54:23Z

I have the environment and have tested it. And the proposed fix is quite simple. Maybe someone else can accept the bug fix.

**Comment by ansibot** [comment]
From @bcoca on 2016-03-04T15:54:23Z

so the code already has similar fix in place, is this still an issue in 2.2?
`if rc not in (None, 0, 2, 10, 20):`

**Comment by ansibot** [comment]
From @ceving on 2016-03-04T15:54:23Z

Sorry have only 2.1.1 in the Solaris environment and can not test 2.2.

**Comment by ansibot** [comment]
From @ansibot on 2016-03-04T15:54:23Z

@dothebart ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dothebart** [comment]
Sorry not using Solaris anymore. However, you should be easily be able to reproduce this bug by the description.

Since it basically makes the package useless, you either want to fix it or delete the package. 

I'm fine with either solution, or if you just don't care - close it.

**Comment by danowar2k** [comment]
Please fix it, this basically kills my playbook at the moment.
Ansible version 2.4.2.0, trying to use playbook on Solaris 11.3.

**Comment by ansibot** [comment]
cc @L2G @dagwieers @dermute @jasperla @jpdasma @scathatheworm @szinck @troy2914 @xen0l
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @fishman @mator
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/packaging/os/svr4pkg.py -> https://galaxy.ansible.com/community/general

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
